esmvaltool/cmorizers/data/formatters/datasets/esacci_soilmoisture.py

```diff
@@ -0,0 +1,191 @@
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+        data cube with fixed coordinates.
+    """
+    # first fix any completely missing coord var names
+    fix_dim_coordnames(cube)
+    # fix individual coords
+    for cube_coord in cube.coords():
+        # fix time
+        if cube_coord.var_name == 'time':
+            logger.info("Fixing time...")
+            cube.coord('time').convert_units(
+                Unit('days since 1970-01-01T00:00:00+00:00',
+                     calendar='proleptic_gregorian'))
+            if overwrite_time_bounds or not cube.coord('time').has_bounds():
+                fix_bounds(cube, cube.coord('time'))
+
+        # fix longitude
+        if cube_coord.var_name == 'lon':
+            logger.info("Fixing longitude...")
+            if cube_coord.ndim == 1:
+                if cube_coord.points[0] < 0. and \
+                        cube_coord.points[-1] < 181.:
+                    cube_coord.points = \
+                        cube_coord.points + 180.
+                    cube.attributes['geospatial_lon_min'] = 0.
+                    cube.attributes['geospatial_lon_max'] = 360.
+                    nlon = len(cube_coord.points)
+                    roll_cube_data(cube, nlon // 2, -1)
+            if overwrite_lon_bounds or not cube_coord.has_bounds():
+                fix_bounds(cube, cube_coord)
+
+        # fix latitude
+        if cube_coord.var_name == 'lat':
+            logger.info("Fixing latitude...")
+            if overwrite_lat_bounds or not cube.coord('latitude').has_bounds():
+                fix_bounds(cube, cube.coord('latitude'))
+
+    return cube
+
+
+def extract_variable(var_info, raw_info, attrs, year):
+    """Extract variables."""
+    rawvar = raw_info['name']
+    constraint = iris.Constraint(name=rawvar)
+    if rawvar == 'sm_uncertainty':
+        sm_cube = iris.load_cube(raw_info['file'],
+                                 iris.NameConstraint(var_name='sm'))
+        ancillary_var = sm_cube.ancillary_variable(
+            'Volumetric Soil Moisture Uncertainty'
+        )
+        cube = sm_cube.copy(ancillary_var.core_data())
+    else:
+        cube = iris.load_cube(raw_info['file'], constraint)
+
+    # Continue with processing the cube if it was successfully loaded
+    fix_var_metadata(cube, var_info)
+    convert_timeunits(cube, year)
+    fix_coords_esacci_soilmoisture(cube, overwrite_time_bounds=False)
+    set_global_atts(cube, attrs)
+
+    # Remove dysfunctional ancillary data without standard names
+    for ancillary_variable in cube.ancillary_variables():
+        cube.remove_ancillary_variable(ancillary_variable)
+
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize data."""
+    glob_attrs = cfg['attributes']
+
+    # run the cmorization
+    for var_name, vals in cfg['variables'].items():
+        all_data_cubes = []
+        if isinstance(vals, dict):  # Ensure vals is a dictionary
```

Comment:
```suggestion
        if not isinstance(vals, dict):  # Ensure vals is a dictionary
             raise ValueError(
                f"Invalid format for variable {var_name}: {type(vals)}"
            )
```

Then do not have to indent the entire part of the code below